Source/RN4UE4/ReplicaRigidDynamicClient.cpp

```diff
@@ -11,27 +19,146 @@ void UReplicaRigidDynamicClient::BeginPlay()
+		data.isEnableCCD = rigid->getRigidBodyFlags().isSet(PxRigidBodyFlag::eENABLE_CCD);
+		data.isEnableCCDFriction = rigid->getRigidBodyFlags().isSet(PxRigidBodyFlag::eENABLE_CCD_FRICTION);
+		data.isEnableCCDMaxContact = rigid->getRigidBodyFlags().isSet(PxRigidBodyFlag::eENABLE_CCD_MAX_CONTACT_IMPULSE);
+		data.isPoseIntegration = rigid->getRigidBodyFlags().isSet(PxRigidBodyFlag::eENABLE_POSE_INTEGRATION_PREVIEW);
+		data.isKinematic = rigid->getRigidBodyFlags().isSet(PxRigidBodyFlag::eKINEMATIC);
+		data.isKinematicScene = rigid->getRigidBodyFlags().isSet(PxRigidBodyFlag::eUSE_KINEMATIC_TARGET_FOR_SCENE_QUERIES);
+		data.isAngX = rigid->getRigidDynamicLockFlags().isSet(PxRigidDynamicLockFlag::eLOCK_ANGULAR_X);
+		data.isAngZ = rigid->getRigidDynamicLockFlags().isSet(PxRigidDynamicLockFlag::eLOCK_ANGULAR_Y);
+		data.isAngY = rigid->getRigidDynamicLockFlags().isSet(PxRigidDynamicLockFlag::eLOCK_ANGULAR_Z);
+		data.isLinX = rigid->getRigidDynamicLockFlags().isSet(PxRigidDynamicLockFlag::eLOCK_LINEAR_X);
+		data.isLinZ = rigid->getRigidDynamicLockFlags().isSet(PxRigidDynamicLockFlag::eLOCK_LINEAR_Y);
+		data.isLinY = rigid->getRigidDynamicLockFlags().isSet(PxRigidDynamicLockFlag::eLOCK_LINEAR_Z);
+		data.restitution = orionMesh->GetBodySetup()->GetPhysMaterial()->GetPhysXMaterial()->getRestitution();
+		PxCombineMode::Enum restituCombineMode = orionMesh->GetBodySetup()->GetPhysMaterial()->GetPhysXMaterial()->getRestitutionCombineMode();
+		data.dynamicFriction = orionMesh->GetBodySetup()->GetPhysMaterial()->GetPhysXMaterial()->getDynamicFriction();
+		data.staticFriction = orionMesh->GetBodySetup()->GetPhysMaterial()->GetPhysXMaterial()->getStaticFriction();
+		PxCombineMode::Enum frictionCombineMode = orionMesh->GetBodySetup()->GetPhysMaterial()->GetPhysXMaterial()->getFrictionCombineMode();
+		//PxFlags<PxMaterialFlag::Enum, PxU16> flags = vismesh->GetStaticMeshComponent()->GetBodySetup()->GetPhysMaterial()->GetPhysXMaterial()->getFlags();
+		if (restituCombineMode == PxCombineMode::eAVERAGE)
+			data.restitutionCombineMode = 0;
+		else if (restituCombineMode == PxCombineMode::eMIN)
+			data.restitutionCombineMode = 1;
+		else if (restituCombineMode == PxCombineMode::eMULTIPLY)
+			data.restitutionCombineMode = 2;
+		else if (restituCombineMode == PxCombineMode::eMAX)
+			data.restitutionCombineMode = 3;
+		if (frictionCombineMode == PxCombineMode::eAVERAGE)
+			data.frictionCombineModeInt = 0;
+		else if (frictionCombineMode == PxCombineMode::eMIN)
+			data.frictionCombineModeInt = 1;
+		else if (frictionCombineMode == PxCombineMode::eMULTIPLY)
+			data.frictionCombineModeInt = 2;
+		else if (frictionCombineMode == PxCombineMode::eMAX)
+			data.frictionCombineModeInt = 3;
+		//	constructionBitstream->Write<PxMaterialFlags>(flags);
+	}
+	SCENE_UNLOCK_READ(SyncScene);
+	orionMesh->SetSimulatePhysics(false);
+}
+
+void UReplicaRigidDynamicClient::GetNearestStaticMesh()
```

Comment:
1. What is this function for?
2. Can we try to avoid so many levels of indentation - e.g. negate your if checks and return or break early instead.
3. The function is called *Get*NearestStaticMesh() - but it's a void function that doesn't have a return value - consider refactoring so that it behaves more like its name suggests.